site/public/js/ta-grading-rubric.js

```diff
@@ -2929,7 +2929,7 @@ function saveComponent(component_id) {
         // The grader unchecked the custom mark, but didn't delete the text.  This shouldn't happen too often,
         //  so prompt the grader if this is what they really want since it will delete the text / score.
         let gradedComponent = getGradedComponentFromDOM(component_id);
-        if (gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
+        if (gradedComponent.custom_mark_enabled && gradedComponent.comment !== '' && !gradedComponent.custom_mark_selected) {
```

Comment:
This variable is also undefined though?

The return from `getGradedComponentFromDOM` is:

https://github.com/Submitty/Submitty/blob/8e7563346c2522014aeb9c38bb689a36dc3dfa46/site/public/js/ta-grading-rubric.js#L1262-L1271

I'm guessing you'll want to modify that function to do an existence check on the custom mark field and return that as a boolean as `custom_mark_enabled` within that function.

Reply:
After defining the variable in `getGradedComponentFromDOM` I noticed a bug that had no doubt been present in my custom-mark fix from the start:
On a gradeable with both peer and non-peer components, opening a non-peer component then opening a peer component allows the custom mark to be selected (and is visible) on peer components. A refresh fixes this.
Although I am dealing with some high-priority assignments at the moment, I hope to commit a fix for this bug in the next few days, as well as the variable definition.

Reply:
@MasterOdin I believe I have fixed these problems, could you take another look ?